### ISSUE TITLE
Add automatic detection of client update

### ISSUE THREAD

**Issue by KUTlime** [mrpond/BlockTheSpot#321]
## Current situation

The installation script asks if user wants to update the Spotify client version. This question is basically irrelevant, because a user has to check currently supported version by BlockTheSpot and check the client version.

## Feature Request 

The installation script can and should detect if there is an installed Spotify version what is the version. If the version is unsupported, the version will be updated automatically.

## Detailed design

Detection is easy

```powershell
(Get-childitem -LiteralPath $env:APPDATA\Spotify\Spotify.exe).VersionInfo
```

Real deal is where to store the supported versions. It should be one place and both the installation script & Readme file should take this store as a source of truth.

### Recommended action breakdown
1. Create a storage for minimal supported version in the repo.
2. Inject the version from the storage into Readme.
3. Implement detection of Spotify version 
4. Remove corresponding `Read-Host` call & input parameter.

**Comment by mrpond** [comment]
I have little concern here,

I live in Asia, and Spotify update slow than other region from few days to week.

Maybe we can talk about unsupported version instead,
e.g. if (Spotify.VersionInfo < xxx.xxx) Then Reject the patch.

**Comment by Nuzair46** [comment]
@mrpond Yes I support your point. We should only update Spotify if the current version being used is not supported by BlockTheSpot. Usually BlockTheSpot works on a range of Spotify version. Right now, Every version since October 2021 is supported by BlockTheSpot. So If the user is using the Spotify between the latest and the oldest supported version, we can ignore the update. 

* Also it should be noted that a new Spotify build can break the patch so, user still needs to be able to manually choose to update. So we need to keep a latest unsupported version also.

* Control flow I propose:
  1 Script checks if current installed Spotify version is between the latest and oldest supported version.
  2 If not continue else jump to 4
  3 Ask user permission whether to update or not (avoid updating to unsupported version)
  4 Install patch and continue normal flow.

cc: @KUTlime

**Comment by KUTlime** [comment]
This sounds like a good solution to me.

One note about the upper range bound: This could be either some specific version (_the latest version most preferably_), or `null` for the current state, where all versions are supported since a version `XYZ`. This could decrees a number of useless commits where somebody bumps the Spotify client version up. On the other hand, the problem would be Readme where `null` must be replaced accordingly.

**Comment by Nuzair46** [comment]
@KUTlime In the readme we can just provide the latest supported version. We might not need to make both readme and script use the same place to check for the versions.

**Comment by KUTlime** [comment]
Yes, this could work. The latest supported version in Readme as one thing, and have a version range in installation script or elsewhere as a second thing.

My suggestion is to keep the version range in the installation script itself, for simplicity.

**Comment by KUTlime** [comment]
So... What is the range of supported version then? 🙂
I will prepare PR for this.

**Comment by Nuzair46** [comment]
Right now, we can use the latest 5 from this pr's https://github.com/mrpond/BlockTheSpot/pulls?q=is%3Apr+is%3Aclosed+Bump+Spotify+version+

**Comment by Davoleo** [comment]
I guess we should also update the readme paragraphs according to this new policy: https://github.com/mrpond/BlockTheSpot#known-issues
